### DISCUSSION TITLE
How do I control the Http Status code and response body of an Http Listener channel?

### DISCUSSION THREAD

**Discussion by jerchap** [nextgenhealthcare/connect#5679] [Q&A] [answered]
I have an http listener channel.  I am expecting json so I set the data types to json.
I need to return a 200 if the json is valid and has the correct schema, otherwise return 500.

I've done a few tests.  I've tested with POSTing an empty body,  a valid empty json object {}, an invalid json object {dfsadfsdfjk} and a valid json object that follows the schema.

If I set the channel response to autogenerate before processing,  I will receive a 200 status code in all tests, even if my source errors out when the json can't be parsed by Mirth.

If I set the channel response to autogenerate after source transform, I will get a 500 status code when the json is malformed, and the channel source errors out because Mirth can't parse the json.

If I set the channel response to a javascript destination, so I can control the response body,  I get a 200 status code in all tests, even if my source errors out when the json can't be parsed by Mirth.  The destination never fires because the message errors at the source.

Each configuration seems to have undesirable consequences in my case, because I want an http status code of 500 if the json can't be parsed by Mirth, but I also want to control the body of the response.  How can I do this?

**Comment by pacmano1** [comment, marked as answer]
Set the inbound data type to raw and try to `JSON.parse` the incoming data in a `try...catch` block and proceed accordingly if it is not valid JSON.  i.e.  set a $c variable to "200" or "500" (a string, not a number!) and use that variable in the response status code field on the source tab.
